https://issues.redhat.com/browse/ACM-5510 Updating object-templates-raw (#5141)
open-cluster-management/rhacm-docs: governance/adv_template_process.adoc

If you need to manually initiate an update, use the special annotation, `policy.open-cluster-management.io/trigger-update`, to indicate changes for the data referenced by the templates. Any change to the special annotation value automatically initiates template processing. Additionally, the latest contents of the referenced resource are read and updated in the policy definition that is propagated for processing on managed clusters. A way to use this annotation is to increment the value by one each time.

[#raw-object-template-processing]
== Object template processing

> Set object templates with a YAML string representation. The `object-template-raw` parameter is an optional parameter that support if-else statements. You can specify conditions in the `object-templates` specification. The following example is defined to add the `species-category: mammal` label  to any ConfigMap in the `default` namespace that has a `name` key equal to `Sea Otter`:

Reviewer: if-else is one example of many functions, so I think mentioning `range` and implying there are more would be good.
```suggestion
Set object templates with a YAML string representation. The `object-template-raw` parameter is an optional parameter that supports advanced templating use-cases, such as if-else and the `range` function. The following example is defined to add the `species-category: mammal` label  to any ConfigMap in the `default` namespace that has a `name` key equal to `Sea Otter`:
```

[source,yaml]
----
object-templates-raw: |
  {{- range (lookup "v1" "ConfigMap" "default" "").items }}
  {{- if eq .data.name "Sea Otter" }}
  - complianceType: musthave
    objectDefinition:
      kind: ConfigMap
      apiVersion: v1
      metadata:
        name: {{ .metadata.name }}
        namespace: {{ .metadata.namespace }}
        labels:
          species-category: mammal
  {{- end }}
  {{- end }}
----

> *Note:* While `spec.object-templates` and `spec.object-templates-raw` are optional, at least one of the two parameter fields must be set. 

Reviewer: ```suggestion
*Note:* While `spec.object-templates` and `spec.object-templates-raw` are optional, one of the two parameter fields must be set.
```

If you need to manually initiate an update, use the special annotation, `policy.open-cluster-management.io/trigger-update`, to indicate changes for the data referenced by the templates. Any change to the special annotation value automatically initiates template processing. Additionally, the latest contents of the referenced resource are read and updated in the policy definition that is propagated for processing on managed clusters. A way to use this annotation is to increment the value by one each time.

[#raw-object-template-processing]
== Object template processing

> Set object templates with a raw YAML string. The `object-template-raw` parameter is an optional parameter that support if-else statements. You can specify conditions `object-templates` specification. The following example is defined to add the `species-category: mammal` label  to any ConfigMap in the `default` namespace that has a `name` key equal to `Sea Otter`:

Reviewer 2: ```suggestion
Set object templates with a YAML string representation. The `object-template-raw` parameter is an optional parameter that support if-else statements. You can specify conditions in the `object-templates` specification. The following example is defined to add the `species-category: mammal` label  to any ConfigMap in the `default` namespace that has a `name` key equal to `Sea Otter`:
```

[source,yaml]
----

> object-templates:

Reviewer 2: You can't specify both object-templates and object-templates-raw.
```suggestion
```

object-templates-raw: |
{{- range (lookup "v1" "ConfigMap" "default" "").items }}
    {{- if eq .data.name "Sea Otter" }}
    - complianceType: musthave
      objectDefinition:
        kind: ConfigMap
        apiVersion: v1
        metadata:
          name: {{ .metadata.name }}
          namespace: {{ .metadata.namespace }}
          labels:
            species-category: mammal

>     {{- end }}

Reviewer 2: This should all be indented.

Author: Updated based on the example that is provided in the issue
